frontend/test/metabase/scenarios/question/reproductions/17910-revision-history-update.cy.spec.js

```diff
@@ -0,0 +1,34 @@
+import { restore, openOrdersTable, modal } from "__support__/e2e/cypress";
+
+describe("issue 17910", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("Revisions should work after creating a question without reloading", () => {
+    openOrdersTable();
+    cy.intercept("POST", `/api/card`).as("card");
+    cy.findByText("Save").click();
+    modal().within(() => {
+      cy.findByText("Save").click();
+    });
+    cy.wait("@card");
+    modal().within(() => {
+      cy.findByText("Not now").click();
+    });
+    cy.findByTestId("saved-question-header-button").click();
+    cy.findByText("Add a description").click();
+    modal().within(() => {
+      cy.get("textarea[name=description]").type("A description");
+      cy.findByText("Save").click();
+    });
+    cy.findByText("History").click();
+    cy.findByText("History")
+      .siblings()
+      .first()
+      .find("ul")
+      .children()
+      .should("to.have.length", 2);
```

Comment:
Tests relying on DOM structure that much can be pretty fragile, ie we can add/remove a few intermediate DOM nodes and it can break the test. We can add `data-testid` attributes to elements and then query them using `findByTestId` as an alternative